tasks/main.yml

```diff
@@ -26,12 +23,16 @@
     url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz'
     dest: /tmp/
   when: not mattermost_enterprise
+  become: false
+  delegate_to: localhost
 
-- name: 'Download Enterprise version binary from https://releases.mattermost.com'
+- name: 'Download enterprise version binary from https://releases.mattermost.com'
   get_url: 
     url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz'
     dest: /tmp/
   when: mattermost_enterprise
+  become: false
+  delegate_to: localhost
 
 - name: Unpack mattermost archive.
   unarchive: 
```

Comment:
I think you need to delegate these tasks to localhost and change the directory to something in `global_cache_dir`.
You can test it by removing the downloaded copy in /tmp on the mattermost server and see if it still extracts to the correct location.